middleware/compress.go

```diff
@@ -21,6 +19,9 @@ type (
 )
 
 func (w gzipWriter) Write(b []byte) (int, error) {
+	if "" == w.Header().Get("Content-Type") {
```

Comment:
- Do we get any benefit with this? If not, lets swap the condition.
- We have a contant `echo.ContentType` for Content-Type.